presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -4376,6 +4376,40 @@ public void testParquetWithMissingColumns()
         assertUpdate(sessionUsingColumnIndex, "DROP TABLE " + multiColumnTableName);
     }
 
+    @Test
+    public void testParquetWithMissingNestedColumns()
+    {
+        Session sessionUsingColumnIndex = Session.builder(getSession())
+                .setCatalogSessionProperty(catalog, "parquet_use_column_names", "true")
+                .build();
+
+        String missingNestedFieldsTableName = "test_parquet_missing_nested_fields";
+
+        assertUpdate(format(
+                "CREATE TABLE %s(" +
+                        "  a ARRAY(ROW(a2 int))) " +
+                        "WITH (format='PARQUET')",
+                missingNestedFieldsTableName));
+        assertUpdate(sessionUsingColumnIndex, "INSERT INTO " + missingNestedFieldsTableName + " VALUES (ARRAY[ROW(2)])", 1);
+
+        String tableLocation = (String) computeActual("SELECT DISTINCT regexp_replace(\"$path\", '/[^/]*$', '') FROM " + missingNestedFieldsTableName).getOnlyValue();
+        String missingNestedArrayTableName = "test_parquet_missing_nested_array";
+        assertUpdate(sessionUsingColumnIndex, format(
+                "CREATE TABLE %s(" +
+                        "  a ARRAY(ROW(a1 ARRAY(varchar), a2 int))) " +
+                        "WITH (format='PARQUET', external_location='%s')",
+                missingNestedArrayTableName,
+                tableLocation));
+
+        assertQuery(
+                sessionUsingColumnIndex,
+                "SELECT a[1].a1 FROM " + missingNestedArrayTableName,
+                "VALUES (null)");
```

Comment:
can we verify the behavior with `sessionUsingColumnName` as well?

also, as this is a regression test, worth recording original NPE failure (can be directly in the commit message)

Reply:
(sorry for the latency)
This update address both points